src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1093,6 +1101,63 @@ private void EmitSearchSelectionTelemetry(PackageItemViewModel selectedPackage)
             }
         }
 
+        private void IncrementInstalledPackageSelectionCount()
+        {
+            PackageItemViewModel selectedItem = _packageList.SelectedItem;
+            if (selectedItem == null || ActiveFilter != ItemFilter.Installed)
+            {
+                return;
+            }
+
+            if (selectedItem.PackageLevel == PackageLevel.TopLevel)
+            {
+                _installedTabTelemetryData.TopLevelPackageSelectedCount++;
+            }
+            else if (selectedItem.PackageLevel == PackageLevel.Transitive)
+            {
+                _installedTabTelemetryData.TransitivePackageSelectedCount++;
+            }
+        }
+
+        private void PackageList_GroupExpansionChanged(object sender, RoutedEventArgs e)
+        {
+            if (sender is Expander expander && expander.Tag is PackageLevel pkgLevel)
```

Comment:
Do we also need to check if we are on Installed Tab?

~~I would also add an UI thread check to avoid double counting when calling on other threads.~~

```suggestion
            if (sender is Expander expander && expander.Tag is PackageLevel pkgLevel && ActiveFilter == ItemFiler.Installed)
```

Reply:
> Do we also need to check if we are on Installed Tab?

We probably should. It's not strictly necessary today since the expanders that will invoke this event handling only appear when the Installed Tab is active, but the extra check might help make the code more readable.

Reply:
> I would also add an UI thread check to avoid double counting when calling on other threads.

@dominoFire: How would it double count if called on other threads? For every call to this method we want to count. If multiple threads call into the same method, each one should increment the number by 1 and the order in which they're incremented doesn't matter. At the end of the day, the only thing that matters is that for every invocation of this method an increment occurs. Can you help me understand how double counting might occur?

Reply:
I multiple background threads runs this method, those can increment the counter multiple times, leading 

Since we are assuming there's only one UI thread, we can enforce to run this code _only_ on UI thread, and, that should avoid double counting in case other non-UI threads run this code.

Reply:
Discussed offline. We don't need UI thread check.